### DISCUSSION TITLE
name ArgMax= column?

### DISCUSSION THREAD

**Discussion by tkawachi** [EvgSkv/logica#82] [Q&A] [answered]
In the following example, I have retrieved `title` with the latest `change_date` for each `id` from `History`.

```
History(id: "A", change_date: Date(2021, 4, 1), title: "A-title2");
History(id: "A", change_date: Date(2021, 3, 1), title: "A-title1");
History(id: "B", change_date: Date(2021, 3, 1), title: "B-title1");

Latest(id:) ArgMax= title -> x :- History(id:, title:, change_date: x);
```

```
+----+--------------+
| id | logica_value |
+----+--------------+
| A  | A-title2     |
| B  | B-title1     |
+----+--------------+
```

Here, title is assigned a column named `logica_value`, but I want to rename it to `title`. How can I do this?
(In fact, I don't know how `Latest(id:) ArgMax= title -> x :- History(id:, title:, change_date: x);` is interpreted grammatically.)

**Comment by EvgSkv** [comment, marked as answer]
The value is simply interpreted as a column named `logica_value`.
To put the result of `ArgMax` to the `title` column we can do as follows.

```
History(id: "A", change_date: Date(2021, 4, 1), title: "A-title2");
History(id: "A", change_date: Date(2021, 3, 1), title: "A-title1");
History(id: "B", change_date: Date(2021, 3, 1), title: "B-title1");

Latest(id:, title? ArgMax= title -> x) distinct :- History(id:, title:, change_date: x);
```

The arrow notation `a -> b` is simply a nice way to write `{arg: a, value: b}` and the `ArgMax` aggregation function takes such record as an argument, returning `arg` which has the largest `value`.

Please let me know if you have further questions.

**Reply by tkawachi** [reply]
Thank you very much. I'm starting to enjoy logica more and more.
